### DISCUSSION TITLE
OpenRefine as tool to identify and repair broken data entries

### DISCUSSION THREAD

**Discussion by tholzheim** [SmartDataAnalytics/OpenResearch#176] [Show and tell]
![Screenshot_2021-04-17 events json - OpenRefine](https://user-images.githubusercontent.com/45004115/115122320-94d8cd80-9fb7-11eb-8308-2b9244b329e5.png)

* Raw data: [or_events.zip](https://github.com/SmartDataAnalytics/OpenResearch/files/6329710/or_events.zip) (data from wiki backup files extracted with wikiextract)
* Blank project: [openresearch_events.openrefine.tar.gz](https://github.com/SmartDataAnalytics/OpenResearch/files/6329715/openresearch_events.openrefine.tar.gz)
* Project with some applied changes: [events-json.openrefine.tar.gz](https://github.com/SmartDataAnalytics/OpenResearch/files/6329756/events-json.openrefine.tar.gz)

## Identify invalid data 
* #164 
* #157

### Example usage for #71
1. Trim whitespace
![Screenshot from 2021-04-17 20-09-47](https://user-images.githubusercontent.com/45004115/115122705-9efbcb80-9fb9-11eb-9f3b-5218d7015ffc.png)
2.  Converte to date
![Screenshot from 2021-04-17 20-10-04](https://user-images.githubusercontent.com/45004115/115122726-b935a980-9fb9-11eb-86b2-e9a59aa0fba4.png)
3.  Select Timeline facet
![Screenshot from 2021-04-17 20-10-28](https://user-images.githubusercontent.com/45004115/115122676-7bd11c00-9fb9-11eb-8b3f-ba505f2dcdaa.png)
4.  Select all entries that could not be converted
![Screenshot from 2021-04-17 20-10-40](https://user-images.githubusercontent.com/45004115/115122760-eb470b80-9fb9-11eb-81fd-3db69b6151dc.png)

**Comment by WolfgangFahl** [comment]
Impressive. I'd love to see the process from the OpenRefine Results to the import. Can we do that as a unix-pipeline already? 

- Input-Query -> list of events/series 
- list of events/series -> PTP -> computer readable signature of events/series (JSON) with provenance ids
- signature lists -> OpenRefine (optional if data is messy or incomplete (e.g. provenance)) -> cleaned up list of signatures (JSON)
- computer readable event / event series list cleaned up with signatures & provenance
- import into wiki via wikirender

the whole above procedure would be related to [Exchange/Transfer Button for trials of import](https://github.com/WolfgangFahl/ProceedingsTitleParser/issues/8)
